src/modules/bootloader/main.py

```diff
@@ -145,43 +127,84 @@ def create_systemd_boot_conf(install_path, efi_dir, uuid, entry, entry_name, ker
         kernel_params.append("resume=/dev/mapper/{!s}".format(
             swap_outer_mappername))
 
-    kernel_line = get_kernel_line(kernel_type)
+    kernel_line = entry_name + " " + kernel_version
     libcalamares.utils.debug("Configure: \"{!s}\"".format(kernel_line))
 
     if kernel_type == "fallback":
-        img = libcalamares.job.configuration["fallback"]
-        entry_name = entry_name + "-fallback"
+        version_string = kernel_version + "-fallback"
+        initrd = "initrd-fallback"
+        mkinitcpio_option = "-S autodetect"
     else:
-        img = libcalamares.job.configuration["img"]
+        version_string = kernel_version
+        initrd = "initrd"
+        mkinitcpio_option = ""
 
-    conf_path = os.path.join(install_path + efi_dir,
-                             "loader",
-                             "entries",
-                             entry_name + ".conf")
+    # get the machine-id
+    with open(os.path.join(install_path, "etc", "machine-id"), 'r') as machineid_file:
+        machine_id = machineid_file.read().rstrip('\n')
```

Comment:
I should add some better error handling there but that wouldn't really resolve the issue since we need the machine-id.

I can add the module dependency if that would be an acceptable solution.